llvm/include/llvm/IR/InstrTypes.h

```diff
@@ -1345,6 +1345,16 @@ class CallBase : public Instruction {
     return nullptr;
   }
 
+  /// Shortcut to retrieving the name of the called function.
+  /// Returns std::nullopt, if the function cannot be found.
+  std::optional<StringRef> getCalledFunctionName() const {
+    Function *F = getCalledFunction();
+    if (F)
+      return F->getName();
+
+    return std::nullopt;
```

Comment:
Can we use an empty StringRef as the sentinel instead?

Reply:
Technically, yes, but I think `std::optional` is a better fit, since using `operator*` on the return value would fail if no value is present.

Reply:
Which is the kind of failure you are trying to avoid needing to worry about?